eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxPublisher.scala

```diff
@@ -75,28 +75,31 @@ object ReplaceableTxPublisher {
     }
   }
 
-  def getFeerate(feeEstimator: FeeEstimator, confirmBefore: BlockHeight, currentBlockHeight: BlockHeight, hasEnoughSafeUtxos: Boolean): FeeratePerKw = {
-    val remainingBlocks = confirmBefore - currentBlockHeight
-    if (hasEnoughSafeUtxos) {
-      val blockTarget = remainingBlocks match {
-        // If our target is still very far in the future, no need to rush
-        case t if t >= 144 => 144
-        case t if t >= 72 => 72
-        case t if t >= 36 => 36
-        // However, if we get closer to the target, we start being more aggressive
-        case t if t >= 18 => 12
-        case t if t >= 12 => 6
-        case t if t >= 2 => 2
-        case _ => 1
-      }
-      feeEstimator.getFeeratePerKw(blockTarget)
-    } else {
-      // We don't have many safe utxos so we want the transaction to confirm quickly.
-      if (remainingBlocks <= 1) {
-        feeEstimator.getFeeratePerKw(1)
-      } else {
-        feeEstimator.getFeeratePerKw(2)
-      }
+  def getFeerate(feerates: FeeratesPerKw, confirmationTarget: ConfirmationTarget, currentBlockHeight: BlockHeight, hasEnoughSafeUtxos: Boolean): FeeratePerKw = {
+    confirmationTarget match {
+      case ConfirmationTarget.Absolute(confirmBefore) =>
+        // If we have an absolute block height target, we take into account what the current height is and adjust the feerate
+        val remainingBlocks = confirmBefore - currentBlockHeight
+        if (hasEnoughSafeUtxos) {
+          remainingBlocks match {
+            // If our target is still very far in the future, no need to rush
+            case t if t >= 36 => feerates.slow
+            // However, if we get closer to the target, we start being more aggressive
+            case t if t >= 12 => feerates.medium
+            case t if t >= 2 => feerates.fast
+            case _ => feerates.fastest
+          }
```

Comment:
This changes the dynamic of the fee bumping and is somewhat tailored to mempool-style fee estimation.